File: Java/Memory.java
public class Memory extends Hardware{

    private int frequency = 0;
    private int memorySize = 0;
    private String supportsCuda = "";

    public Memory(int frequency, int memorySize, String supportsCuda) {
        this.frequency = frequency;
        this.memorySize = memorySize;
        this.supportsCuda = supportsCuda;
    }

    public Memory() {

    }

    public void setFrequency(int frequency) {
        this.frequency = frequency;
    }
    public int getFrequency() {
        return this.frequency;
    }

    public void setMemorySize(int memorySize) {
        this.memorySize = memorySize;
    }
    public int getMemorySize() {
        return this.memorySize;
    }

    public void setSupportsCuda(String supportsCuda) {
        this.supportsCuda = supportsCuda;
    }
    public String getSupportsCuda() {
        return this.supportsCuda;
    }

}

File: CPP/Memory.cpp
//kelas yang digunakan untuk mengimplementasikan kelas Memory
class Memory : public Hardware
{
	// private attributes
	private:
		// private attributes
		int frequency;
		int memorySize;
		string supportsCuda;

	// public attributes
	public:
		// constructor with parameter
		Memory(int frequency, int memorySize, string supportsCuda)
		{
			this->frequency = frequency;
			this->memorySize = memorySize;
			this->supportsCuda = supportsCuda;
		}
		// constructor
		Memory()
		{
			this->frequency = 0;
			this->memorySize = 0;
			this->supportsCuda = "";
		}

		// method for frequency attribute
		void setFrequency(int frequency)
		{
			this->frequency = frequency;
		}
		int getFrequency()
		{
			return this->frequency;
		}

		// method for memory size attribute
		void setMemorySize(int memorySize)
		{
			this->memorySize = memorySize;
		}
		int getMemorySize()
		{
			return this->memorySize;
		}

		// method for supports cuda attribute
		void setSupportsCuda(string supportsCuda)
		{
			this->supportsCuda = supportsCuda;
		}
		string getSupportsCuda()
		{
			return this->supportsCuda;
		}

		// destructor
		~Memory(){

		}

};